astropy/units/utils.py

```diff
@@ -199,7 +196,34 @@ def sanitize_scale(scale):
         return scale.real
 
 
-def validate_power(p, support_tuples=False):
+def maybe_simple_fraction(p, max_denominator=100):
+    """Fraction very close to x with denominator at most max_denominator.
+
+    The fraction has to be such that fraction/x is unity to within 4 ulp.
+    If such a fraction does not exist, returns the float number.
+
+    The algorithm is that of `fractions.Fraction.limit_denominator`, but
+    sped up by not creating a fraction to start with.
+    """
+    if p == 0 or p.__class__ is int:
+        return p
+    n, d = p.as_integer_ratio()
+    a = n // d
+    # Normally, start with 0,1 and 1,0; here we have applied first iteration.
+    n0, d0 = 1, 0
+    n1, d1 = a, 1
+    while d1 <= max_denominator:
```

Comment:
Will there be a case where this loop would effectively run forever? Do we need a `max_iter`?

Reply:
Okie dokie then... :smile_cat: